src/components/Carousel.js

```diff
@@ -1,19 +1,154 @@
+    });
+  }
 
-export default Carousel;
+  goNext = () => {
+    const { offset, frameSize, step, itemWidth, infinite }
+    = this.state;
+    const maxOffset = -1 * (10 - frameSize) * itemWidth;
+
+    if (infinite) {
+      this.setState({ disabledPrev: false });
+      if (offset === maxOffset) {
+        this.setState({ offset: 0 });
+      } else if (offset + -1 * itemWidth * step < maxOffset) {
+        this.setState({ offset: maxOffset });
+      } else {
+        this.setState(state => (
+          { offset: state.offset - state.itemWidth * state.step }
+        ));
+      }
+    } else {
+      this.setState({ disabledPrev: false });
+      if (offset + -1 * itemWidth * step <= maxOffset) {
+        this.setState({ offset: maxOffset });
+        this.setState({ disabledNext: true });
+      } else {
+        this.setState(state => (
+          { offset: state.offset - state.itemWidth * state.step }
+        ));
+      }
+    }
+  }
+
+  goPrev = () => {
+    const { offset, frameSize, step, itemWidth, infinite }
+    = this.state;
+    const maxOffset = -1 * (10 - frameSize) * itemWidth;
+
+    if (infinite) {
+      this.setState({ disabledNext: false });
+      if (offset === 0) {
+        this.setState({ offset: maxOffset });
+      } else if (offset + itemWidth * step > 0) {
+        this.setState({ offset: 0 });
+      } else {
+        this.setState(state => (
+          { offset: state.offset + state.itemWidth * state.step }
+        ));
+      }
+    } else {
+      this.setState({ disabledNext: false });
+      if (offset >= -1 * (itemWidth * step)) {
+        this.setState({ offset: 0 });
+        this.setState({ disabledPrev: true });
+      } else {
+        this.setState(state => (
+          { offset: state.offset + state.itemWidth * state.step }
+        ));
+      }
+    }
+  }
+
+  render() {
+    const { offset, frameSize, itemWidth, animationDuration,
+      allImages, disabledPrev, disabledNext }
+      = this.state;
+
+    return (
+      <div className="Carousel">
+        <Form
+          {...this.props}
+          updateData={this.updateData}
+        />
+        <div
+          className="Page"
+          style={{ width: itemWidth * frameSize }}
+        >
+          <ul
+            className="Carousel__list"
+            style={{
+              transform: `translateX(${offset}px)`,
+              transition: `${animationDuration}ms`,
+            }}
+          >
+            {allImages.map(item => (
+              <li key={allImages.indexOf(item)}>
+                <img
+                  style={{ width: `${itemWidth}px` }}
+                  src={item}
+                  alt={allImages.indexOf(item)}
+                />
+              </li>
+            ))}
+          </ul>
+        </div>
+        <div className="buttons">
+          <button
+            className="button is-success is-rounded"
+            disabled={disabledPrev}
+            type="button"
+            onClick={() => this.goPrev()}
```

Comment:
якщо метод не приймає параметрів то стрілочну функцію не потрібно писати
```suggestion
            onClick={this.goPrev}
```